zmon-controller-app/src/main/java/org/zalando/zmon/service/impl/ZMonServiceImpl.java

```diff
@@ -291,6 +291,56 @@ public void deleteDetachedCheckDefinitions() {
         return checkResults;
     }
 
+    @Override
+    public List<CheckResults> getCheckResultsWithoutEntities(final int checkId, final String entity, final int limit) {
+        final List<ResponseHolder<String, List<String>>> results = new LinkedList<>();
+        final List<ResponseHolder<Integer, Long>> alertEntitiesCount = new LinkedList<>();
+        final List<ResponseHolder<Integer, Set<String>>> alertEntities = new LinkedList<>();
+
+        final List<Integer> alertDefinitionIds = alertDefinitionSProc.getAlertIdsByCheckId(checkId);
+        try (Jedis jedis = redisPool.getResource()) {
+            final Set<String> entities = (entity == null ? jedis.smembers(RedisPattern.checkEntities(checkId))
+                    : Collections.singleton(entity));
+            if (!entities.isEmpty()) {
+
+                // execute async calls
+                final Pipeline p = jedis.pipelined();
+                for (final String key : entities) {
+                    results.add(
+                            ResponseHolder.create(key, p.lrange(RedisPattern.checkResult(checkId, key), 0, limit - 1)));
+                }
+
+                for (final Integer alertDefinitionId : alertDefinitionIds) {
+                    alertEntitiesCount.add(ResponseHolder.create(alertDefinitionId,
+                            p.scard(RedisPattern.alertEntities(alertDefinitionId))));
+                }
+
+                p.sync();
+            }
+        }
+
+        final List<CheckResults> checkResults = buildCheckResults(results);
+
+        // group all ids by entity
+        final SetMultimap<String, Integer> entities = HashMultimap.create();
+        final Map<Integer, Long> entitiesCount = new HashMap<>();
+
+        for (final ResponseHolder<Integer, Set<String>> response : alertEntities) {
```

Comment:
`alertEntities` is always empty. Please delete this loop and initialisation of `alertEntities` at line 298

Reply:
Updated the code as per above comment